python/ray/util/client/server/dataservicer.py

```diff
@@ -25,7 +25,7 @@ def __init__(self, basic_service: "RayletServicer"):
         self.clients_lock = Lock()
         self.num_clients = 0  # guarded by self.clients_lock
 
-    def Datapath(self, request_iterator, context):
+    async def Datapath(self, request_iterator, context):
```

Comment:
can we split this function a bit for readability? there are plenty of nested ifs, and returns/breaks/continues in many places. I can't understand what this is doing without significant effort.

Reply:
@AmeerHajAli I'd really prefer to push this into a later PR. This is essentially just a `large switch` statement over the possible request types.

If you want, I can move more of the request logic out, but I wanted to further nest it in this **main function** to make it clear that the `continue` logic _only ever happens_ for asynchronous requests.